pkg/kubelet/lifecycle/handlers.go

```diff
@@ -113,21 +121,46 @@ func (hr *HandlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 		}
 		host = status.IP
 	}
-	var port int
-	if handler.HTTPGet.Port.Type == intstr.String && len(handler.HTTPGet.Port.StrVal) == 0 {
-		port = 80
```

Comment:
why did you remove the defaulting?

Reply:
> why did you remove the defaulting?

Because this block of codes will never be run. Since apiserver will do port validation where `port=0` is invalid.

https://github.com/kubernetes/kubernetes/blob/0048d2da400b8c48ae83acc6a223a65f3551674a/pkg/apis/core/validation/validation.go#L2417-L2432

Reply:
Oh, right. Forgot static pods.